src/main/test/at/ac/fhcampuswien/fhmdb/models/MovieTest.java

```diff
@@ -0,0 +1,77 @@
+
+import org.junit.jupiter.api.*;
+
+public class MovieTest {
+
+    @Nested
+    public class CompareTo {
+        static Movie movie1;
+        static Movie movie2;
+        static Movie movie3;
+        static Movie movie4;
+        static Movie movie5;
+
+        @BeforeAll
+        public static void beforeAll() {
+            movie1 = new Movie("Die Hard", "A movie about a hard death.");
+            movie2 = new Movie("Die Easy", "A movie about an easy death.");
+            movie3 = new Movie("A", "An A class movie!");
+            movie4 = new Movie("Z", "A Z class movie!");
+            movie5 = new Movie("", "");
+        }
+
+        @Test
+        public void Movie_A_is_smaller_than_Movie_B_if_the_title_is_alphabetically_first() {
+            Assertions.assertTrue(movie1.compareTo(movie4) < 0);
+
+            Assertions.assertTrue(movie2.compareTo(movie1) < 0);
+            Assertions.assertTrue(movie2.compareTo(movie4) < 0);
+
+            Assertions.assertTrue(movie3.compareTo(movie1) < 0);
+            Assertions.assertTrue(movie3.compareTo(movie2) < 0);
+            Assertions.assertTrue(movie3.compareTo(movie4) < 0);
+
+            Assertions.assertTrue(movie5.compareTo(movie1) < 0);
+            Assertions.assertTrue(movie5.compareTo(movie2) < 0);
+            Assertions.assertTrue(movie5.compareTo(movie3) < 0);
+            Assertions.assertTrue(movie5.compareTo(movie4) < 0);
+        }
+
+        @Test
+        public void Movie_A_is_greater_than_Movie_B_if_the_title_is_alphabetically_last() {
+            Assertions.assertTrue(movie4.compareTo(movie1) > 0);
+
+            Assertions.assertTrue(movie1.compareTo(movie2) > 0);
+            Assertions.assertTrue(movie4.compareTo(movie2) > 0);
+
+            Assertions.assertTrue(movie1.compareTo(movie3) > 0);
+            Assertions.assertTrue(movie2.compareTo(movie3) > 0);
+            Assertions.assertTrue(movie4.compareTo(movie3) > 0);
+
+            Assertions.assertTrue(movie1.compareTo(movie5) > 0);
+            Assertions.assertTrue(movie2.compareTo(movie5) > 0);
+            Assertions.assertTrue(movie3.compareTo(movie5) > 0);
+            Assertions.assertTrue(movie4.compareTo(movie5) > 0);
+        }
+
+        @Test
+        public void Movie_A_is_at_the_same_position_as_Movie_B_if_the_title_is_alphabetically_the_same() {
+            Assertions.assertEquals(0, movie1.compareTo(movie1));
+            Assertions.assertEquals(0, movie2.compareTo(movie2));
+            Assertions.assertEquals(0, movie3.compareTo(movie3));
+            Assertions.assertEquals(0, movie4.compareTo(movie4));
+            Assertions.assertEquals(0, movie5.compareTo(movie5));
+        }
+
+        @Test
+        public void throws_an_exception_if_Movie_B_is_null() {
+            Assertions.assertThrows(IllegalArgumentException.class, () -> movie1.compareTo(null));
+            Assertions.assertThrows(IllegalArgumentException.class, () -> movie2.compareTo(null));
```

Comment:
I think it's enough to do only one comparisson here

Reply:
@sorinlzr solved with the next commit.